src/components/Tables/Filters.tsx

```diff
@@ -0,0 +1,189 @@
+    headers: TableHeaders<Data>,
+    key: Key
+): string => {
+    return headers.find(([headerKey]) => headerKey === key)?.[1] ?? key.toString();
+};
+
+const TextFilterStyling = styled.div`
+    & input[type="text"].MuiInputBase-input.MuiOutlinedInput-input {
+        border: none;
+    }
+`;
+
+interface TextFilterProps<Data, Key extends keyof Data> {
+    key: Key;
+    label: string;
+    caseSensitive?: boolean;
+    initialValue?: string;
+}
+
+const defaultToString = (value: unknown): string => {
+    if (typeof value === "string") {
+        return value;
+    }
+
+    return JSON.stringify(value);
+};
+
+export const textFilter = <Data, Key extends keyof Data>({
+    key,
+    label,
+    caseSensitive = false,
+    initialValue = "",
+}: TextFilterProps<Data, Key>): KeyedFilter<Data, Key, string> => {
+    return keyedFilter(key, label, {
+        state: initialValue,
+        initialState: initialValue,
+        shouldFilter: (data, state) => {
+            let string = defaultToString(data[key]);
+
+            if (!caseSensitive) {
+                string = string.toLowerCase();
+                state = state.toLowerCase();
+            }
+
+            return !string.includes(state);
+        },
+        filterComponent: (state, setState) => {
+            return (
+                <TextFilterStyling>
+                    <FreeFormTextInput
+                        key={label}
+                        value={state}
+                        label={label}
+                        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                            setState(event.target.value);
+                        }}
+                    />
+                </TextFilterStyling>
+            );
+        },
+        clear: (_oldState, setState) => {
+            setState(initialValue);
+        },
+    });
+};
+
+const DateFilterContainer = styled.div`
+    display: flex;
+    gap: 1rem;
+    flex-wrap: wrap;
+
+    & input[type="text"].MuiInputBase-input.MuiOutlinedInput-input {
+        border: none;
+    }
+
+    & .MuiFormControl-root,
+    & .MuiInputBase-root {
+        width: 12rem;
+    }
+`;
+
+interface DateFilterState {
+    from: Date | null;
+    to: Date | null;
+}
+
+interface DateFilterProps<Data, Key extends keyof Data> {
+    key: Key;
+    label: string;
+    initialValue?: DateFilterState;
+}
+
+export const dateFilter = <Data, Key extends keyof Data>({
+    key,
+    label,
+    initialValue = {
+        from: null,
+        to: null,
+    },
+}: DateFilterProps<Data, Key>): KeyedFilter<Data, Key, DateFilterState> => {
+    return keyedFilter(key, label, {
+        state: initialValue,
+        initialState: initialValue,
+        shouldFilter: (data, state) => {
+            const date = data[key] as Date;
+            const from = state.from;
+            const to = state.to;
+
+            if (from !== null && date < from) {
+                return true;
```

Comment:
is 'true' meaning 'it should be filtered [out]' rather than 'the filter should remove all falses'?

Reply:
`true` means it `shouldFilter` i.e. it shouldn't be included